pkg/services/provisioning/apps/types.go

```diff
@@ -0,0 +1,57 @@
+package apps
+
+import "github.com/grafana/grafana/pkg/services/provisioning/values"
+
+// appsAsConfig is normalized data object for apps config data. Any config version should be mappable
+// to this type.
+type appsAsConfig struct {
+	Apps []*appFromConfig
+}
+
+type appFromConfig struct {
+	OrgID          int64
+	OrgName        string
+	PluginID       string
+	Enabled        bool
+	Pinned         bool
+	PluginVersion  string
+	JSONData       map[string]interface{}
+	SecureJSONData map[string]string
+}
+
+type appFromConfigV0 struct {
+	OrgID          values.Int64Value     `json:"org_id" yaml:"org_id"`
+	OrgName        values.StringValue    `json:"org_name" yaml:"org_name"`
+	Type           values.StringValue    `json:"type" yaml:"type"`
+	Disabled       values.BoolValue      `json:"disabled" yaml:"disabled"`
+	JSONData       values.JSONValue      `json:"jsonData" yaml:"jsonData"`
+	SecureJSONData values.StringMapValue `json:"secureJsonData" yaml:"secureJsonData"`
+}
+
+// appsAsConfigV0 is mapping for zero version configs. This is mapped to its normalised version.
+type appsAsConfigV0 struct {
+	Apps []*appFromConfigV0 `json:"apps" yaml:"apps"`
+}
+
+// mapToNotificationFromConfig maps config syntax to normalized notificationsAsConfig object. Every version
+// of the config syntax should have this function.
+func (cfg *appsAsConfigV0) mapToAppFromConfig() *appsAsConfig {
```

Comment:
I think mapToAppsFromConfig is more correct, since it returns apps in plural.
```suggestion
func (cfg *appsAsConfigV0) mapToAppsFromConfig() *appsAsConfig {
```